calculator.js

```diff
@@ -0,0 +1,147 @@
+const ApiUrl = 'https://api.alephium-pool.com';
+
+function statsApiCall(action) {
+    return fetch(`${ApiUrl}${action}`)
+        .then(response => response.json())
+}
+
+function fetchPoolProfit() {
+    return statsApiCall('/profit');
+}
+
+function fetchRate() {
+    return statsApiCall(`/rate`)
+}
+
+function getPoolProfitUSD(rate, profit) {
+    return profitUSD = profit * rate
+}
+
+function perHour(value) {
+    return (value / 24)
+}
+
+function costsPerTime(power_consumption, electricity_costs, multiplier = 1) {
+    return power_consumption * multiplier / 1000 * electricity_costs;
+}
+
+function perWeek(value) {
+    return (value * 7)
+}
+
+function generateHTML(elements_id, periods) {
+    let tbody = document.getElementsByTagName('tbody')[0];
+
+    for (let i = 0; i < 3; i++) {
+        let td = tbody.insertRow();
+        for (let i = 0; i < 4; i++) {
+            if (i === 0) {
+                let tr = td.insertCell();
+                let text = document.createTextNode(periods.shift());
+                tr.appendChild(text);
+            }
+            let tr = td.insertCell();
+            tr.setAttribute('id', elements_id.shift());
+        }
+    }
+}
+
+function addValue(value, element_id, currency_value = null) {
+
+    const costs = ['1h_costs', '24h_costs', '7d_costs']
+    if (costs.includes(element_id)) {
+        document.getElementById(element_id).textContent = `- ${parseFloat(value).toFixed(4)}` + ` ${currency_value}`
+    } else {
```

Comment:
ты можешь распаковать аргументы

```suggestion
    Promise.all([fetchRate(), fetchPoolProfit()]).then([rate, profit]=> {
```

Reply:
Я пробовал так, во первых оно подчеркивает красным, не хватает скобок, а во вторых оно почему то выдает NaN при такое распаковке

Reply:
да, нужно ([rate, profit] => {}) ошибся.